charts/scalardb-analytics-server/templates/scalardb-analytics-server/validate-values.yaml

```diff
@@ -17,3 +17,9 @@
     {{- fail "ScalarDB Analytics Server: \"scalarDbAnalyticsServer.tls.privateKeySecret\" must be set when using user-provided TLS certs (when \"scalarDbAnalyticsServer.tls.enabled\" is \"true\" and \"scalarDbAnalyticsServer.tls.certManager.enabled\" is \"false\")." }}
   {{- end }}
 {{- end }}
+
+{{- if .Values.scalarDbAnalyticsServer.podAnnotations }}
+  {{- if hasKey .Values.scalarDbAnalyticsServer.podAnnotations "checksum/config" }}
+    {{- fail "ScalarDB Analytics Server: \"scalarDbAnalyticsServer.podAnnotations\" cannot contain the key \"checksum/config\" as it is managed by the chart." }}
+  {{- end }}
+{{- end }}
```

Comment:
This chart automatically sets (manages) specific named keys. We want to prevent to override such keys by users. So, I added validations for them.

```
$ helm template test ~/github.com/scalar-labs/helm-charts/charts/scalardb-analytics-server/ -f ./pod-labels-test.yaml
Error: execution error at (scalardb-analytics-server/templates/scalardb-analytics-server/validate-values.yaml:23:8): ScalarDB Analytics Server: "scalarDbAnalyticsServer.podAnnotations" cannot contain the key "checksum/config" as it is managed by the chart.

Use --debug flag to render out invalid YAML
```